pkg/systemd/overview-cards/uncleanShutdownStatus.jsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.js";
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
+import { Icon } from "@patternfly/react-core/dist/esm/components/Icon/index.js";
+import { ExclamationTriangleIcon } from "@patternfly/react-icons";
+
+import cockpit from "cockpit";
+
+import './uncleanShutdownStatus.scss';
+import { useInit } from 'hooks';
+
+const _ = cockpit.gettext;
+
+export const UncleanShutdownStatus = () => {
+    const [uncleanShutdownId, setUncleanShutdownId] = useState(null);
+    const [uncleanShutdownStatusVisible, setUncleanShutdownStatusVisible] = useState(false);
+
+    useInit(() => {
+        cockpit.spawn(
+            ["last", "--system", "--limit=2", "--time-format=iso", "shutdown", "reboot"],
+            { environ: ["LC_ALL=C"], err: "message", }
+        ).then((data) => {
+            const previous_boot = data.split("\n")[1];
+            if (previous_boot === undefined)
+                return;
+
+            // "crash" on wtmpdb, "still running" on util-linux
+            if (!previous_boot.includes("crash") && !previous_boot.includes("still running"))
+                return;
+
+            const lines = previous_boot.split(/ +/);
+            const started = new Date(lines[4]);
+            if (isNaN(started)) {
+                console.warn("cannot parse start date of last line", previous_boot);
+            } else {
+                const epoch = started.getTime().toString();
+                setUncleanShutdownId(epoch);
+                setUncleanShutdownStatusVisible(epoch != cockpit.localStorage.getItem("dismissed-unclean-shutdown-id"));
+            }
+        });
+    });
+
+    if (!uncleanShutdownId || !uncleanShutdownStatusVisible) {
+        return null;
+    }
+
+    function hideAlert() {
+        setUncleanShutdownStatusVisible(false);
+        cockpit.localStorage.setItem('dismissed-unclean-shutdown-id', uncleanShutdownId);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22289-ddb39e90-20251112-150302-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)